### PR TITLE
Extend Article resource

### PR DESCRIPTION
**Pull request by iranl**
- Adds listAll() function to Articles
- Adds results/messages to Articles through BaseObject
- Adds freetext 2 and 3 to Sub Articles
- Removes performancetype and allowchangeperformancetype from Articles
- Changes looping through Sub Article lines from:
```php
$linesDOM->getElementsByTagName('line')
```
To:

```php
$linesDOM->childNodes
``` 

- Some formatting/order changes

### CODE REVIEW

File: src/ApiConnectors/ArticleApiConnector.php [hunk 1]
```diff
@@ -91,4 +88,57 @@ public function sendAll(array $articles): MappedResponseCollection
...
+
+        $response = $this->getFinderService()->searchFinder(FinderService::TYPE_ITEMS, $pattern, $field, $firstRow, $maxRows, $optionsArrayOfString);
+
+        if ($response->data->TotalRows == 0) {
+            return [];
+        }
+
+        $articles = [];
+
+        foreach ($response->data->Items->ArrayOfString as $articleArray) {
```
Comment (willemstuursma) [inline src/ApiConnectors/ArticleApiConnector.php]: I see this bit of code again and again in the PRs. Should this be moved to a private method or a helper trait?

File: src/Article.php [hunk 2]
```diff
@@ -9,179 +9,154 @@
...
-    public function setShortName($shortName)
+    public function setAllowDecimalQuantity(bool $allowDecimalQuantity): self
     {
-        $this->shortName = $shortName;
+        $this->allowDecimalQuantity = $allowDecimalQuantity;
         return $this;
     }
 
-    public function getUnitNameSingular()
+    public function getAllowDiscountorPremium(): bool
```
Comment (willemstuursma) [inline src/Article.php]: Typo, Or is not camel cased.

File: src/Article.php [hunk 3]
```diff
@@ -9,179 +9,154 @@
...
+    public function setShortName($shortName)
     {
-        $this->allowChangePerformanceType
-            = $allowChangePerformanceType;
+        $this->shortName = $shortName;
         return $this;
     }
 
-    public function getPercentage()
+    public function getStatus()
```
Comment (willemstuursma) [inline src/Article.php]: Can this be changed to an Enum

File: src/Article.php [hunk 4]
```diff
@@ -9,179 +9,154 @@
...
-    public function setPercentage($percentage)
+    public function setStatus($status)
     {
-        $this->percentage = $percentage;
+        $this->status = $status;
         return $this;
     }
 
-    public function getAllowDiscountorPremium(): bool
+    public function getType()
```
Comment (willemstuursma) [inline src/Article.php]: Can this be changed to an Enum?

File: src/ArticleLine.php [hunk 5]
```diff
@@ -8,85 +8,76 @@
  */
 class ArticleLine
 {
+    private $freeText1;
```
Comment (willemstuursma) [inline src/ArticleLine.php]: Didn't we have some traits for FreeText?

File: src/ArticleLine.php [hunk 6]
```diff
@@ -112,6 +103,17 @@ public function setShortName($shortName)
         return $this;
     }
 
+    public function getStatus()
```
Comment (willemstuursma) [inline src/ArticleLine.php]: Maybe introduce ArticleStatusTrait?

File: src/ArticleLine.php [hunk 7]
```diff
@@ -123,14 +125,36 @@ public function setSubCode($subCode)
...
 
-    public function setFreeText1($freeText1)
+    public function setUnits($units)
     {
-        $this->freeText1 = $freeText1;
+        $this->units = $units;
+        return $this;
+    }
+
+    public function getUnitsPriceExcl()
```
Comment (willemstuursma) [inline src/ArticleLine.php]: Can you make this Money? Maybe move to a trait.

**Comment by iranl** [comment]
> Would it be possible to add / extend some test cases, with actual responses and tried requests to the Twinfield API?

Thanks for the review and requested changes, will get on them in the coming week.

File: src/ApiConnectors/ArticleApiConnector.php (hunk 1 above)
Comment (iranl) [reply src/ApiConnectors/ArticleApiConnector.php]: Fixed in #142

File: src/Article.php (hunk 2 above)
Comment (iranl) [reply src/Article.php]: Fixed in #142

File: src/Article.php (hunk 3 above)
Comment (iranl) [reply src/Article.php]: Fixed in #142

File: src/ArticleLine.php (hunk 5 above)
Comment (iranl) [reply src/ArticleLine.php]: Fixed in #142

File: src/ArticleLine.php (hunk 6 above)
Comment (iranl) [reply src/ArticleLine.php]: Fixed in #142

File: src/ArticleLine.php (hunk 7 above)
Comment (iranl) [reply src/ArticleLine.php]: Fixed in #142

**Comment by iranl** [comment]
Superseded by #142

File: src/Article.php (hunk 4 above)
Comment (iranl) [reply src/Article.php]: Fixed in #142

**Pull request closed without merging**